java/google/registry/rde/imports/RdeContactReader.java

```diff
@@ -80,18 +83,23 @@ public RdeContactReader(
   }
 
   @Override
-  public ContactResource next() throws IOException {
+  public JaxbFragment<XjcRdeContactElement> next() throws IOException {
     if (count < maxResults) {
-      if (parser == null) {
-        parser = newParser();
-        if (parser.isAtContact()) {
+      try {
+        if (parser == null) {
+          parser = newParser();
+          if (parser.isAtContact()) {
+            count++;
```

Comment:
Correct me if I am wrong, but I think the only thing in this block that throws XmlException is readContact(). If that's so, move the try/catch into the helper method, here and in the other files with this pattern.